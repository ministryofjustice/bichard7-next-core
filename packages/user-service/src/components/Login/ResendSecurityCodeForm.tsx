import Form from "components/Form"
import Paragraph from "components/Paragraph"
import Button from "components/Button"

interface ResendSecurityCodeFormProps {
  csrfToken: string
  emailAddress: string | undefined
  stageName: "loginStage" | "resetStage"
  stageValue: "resetSecurityCode"
}

const ResendSecurityCodeForm: React.FC<ResendSecurityCodeFormProps> = ({
  csrfToken,
  emailAddress,
  stageName,
  stageValue
}) => (
  <>
    <h1 className="govuk-heading-xl govuk-!-margin-bottom-7">{"Get a security code"}</h1>
    <Form method="post" csrfToken={csrfToken}>
      <Paragraph>
        {"We will send a code to: "}
        <b>{emailAddress}</b>
      </Paragraph>
      <Paragraph>
        {`Your code can take up to 5 minutes to arrive. Check your spam folder if you don't get an email.`}
      </Paragraph>
      <Paragraph className="govuk-!-padding-bottom-4">{`The code will expire after 30 minutes.`}</Paragraph>
      <input id="email" name="emailAddress" type="hidden" value={emailAddress} />
      <input type="hidden" name={stageName} value={stageValue} />
      <Button id="security-code-button">{"Get security code"}</Button>
    </Form>
  </>
)

export default ResendSecurityCodeForm
