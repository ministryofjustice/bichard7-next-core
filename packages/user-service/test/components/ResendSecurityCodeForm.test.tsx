import { render } from "@testing-library/react"
import ResendSecurityCodeForm from "components/Login/ResendSecurityCodeForm"

describe("ResendSecurityCodeForm", () => {
  const commonProps = {
    csrfToken: "mock-csrf-token",
    emailAddress: "bichard01@example.com",
    stageValue: "resetSecurityCode" as const
  }

  it("should render correctly for 'loginStage'", () => {
    const props = {
      ...commonProps,
      stageName: "loginStage" as const
    }
    const { container } = render(<ResendSecurityCodeForm {...props} />)

    expect(container).toMatchSnapshot()
  })

  it("should render correctly for 'resetStage'", () => {
    const props = {
      ...commonProps,
      stageName: "resetStage" as const
    }
    const { container } = render(<ResendSecurityCodeForm {...props} />)

    expect(container).toMatchSnapshot()
  })
})
