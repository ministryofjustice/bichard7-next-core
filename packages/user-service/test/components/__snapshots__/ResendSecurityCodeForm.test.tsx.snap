// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`ResendSecurityCodeForm should render correctly for 'loginStage' 1`] = `
<div>
  <h1
    class="govuk-heading-xl govuk-!-margin-bottom-7"
  >
    Get a security code
  </h1>
  <form
    method="post"
  >
    <input
      name="CSRFToken"
      type="hidden"
      value="mock-csrf-token"
    />
    <p
      class="govuk-body"
    >
      We will send a code to: 
      <b>
        bichard01@example.com
      </b>
    </p>
    <p
      class="govuk-body"
    >
      Your code can take up to 5 minutes to arrive. Check your spam folder if you don't get an email.
    </p>
    <p
      class="govuk-body govuk-!-padding-bottom-4"
    >
      The code will expire after 30 minutes.
    </p>
    <input
      id="email"
      name="emailAddress"
      type="hidden"
      value="bichard01@example.com"
    />
    <input
      name="loginStage"
      type="hidden"
      value="resetSecurityCode"
    />
    <button
      class="govuk-button"
      data-module="govuk-button"
      id="security-code-button"
      type="submit"
    >
      Get security code
    </button>
  </form>
</div>
`;

exports[`ResendSecurityCodeForm should render correctly for 'resetStage' 1`] = `
<div>
  <h1
    class="govuk-heading-xl govuk-!-margin-bottom-7"
  >
    Get a security code
  </h1>
  <form
    method="post"
  >
    <input
      name="CSRFToken"
      type="hidden"
      value="mock-csrf-token"
    />
    <p
      class="govuk-body"
    >
      We will send a code to: 
      <b>
        bichard01@example.com
      </b>
    </p>
    <p
      class="govuk-body"
    >
      Your code can take up to 5 minutes to arrive. Check your spam folder if you don't get an email.
    </p>
    <p
      class="govuk-body govuk-!-padding-bottom-4"
    >
      The code will expire after 30 minutes.
    </p>
    <input
      id="email"
      name="emailAddress"
      type="hidden"
      value="bichard01@example.com"
    />
    <input
      name="resetStage"
      type="hidden"
      value="resetSecurityCode"
    />
    <button
      class="govuk-button"
      data-module="govuk-button"
      id="security-code-button"
      type="submit"
    >
      Get security code
    </button>
  </form>
</div>
`;
